### PR TITLE
[OpenAPIGenerator] Add section about issues during `anyOf/oneOf`

### PR DESCRIPTION
**Pull request by KavithaSiva**
## What Has Changed?

Added a section explaining when compilation error occurs while input specification contains keywords `anyOf/oneOf`

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] ~You verified all new and changed links still work (changing the `id` or name of a file can break links)~
- [x] ~You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~

### CODE REVIEW

**Review by newtork — APPROVED** [review]
LGTM

I would be happy about an option to collapse the YAML content.

**Comment by KavithaSiva** [comment]
> LGTM
> 
> I would be happy about an option to collapse the YAML content.

I think Matthias used something like that in the K8s documentation. Let me look that up.

**Comment by KavithaSiva** [comment]
@artemkovalyov Can you please also suggest if this is the right place to document this or if you want it under the Troubleshooting section?

File: docs/java/features/rest/generate-rest-client.mdx [hunk 1]
```diff
@@ -237,3 +237,103 @@ The complete list of available parameters with their description is as follows:
 
 </TabItem>
 </Tabs>
+
+## Solving Compilation Issues After Generation
+
+:::info
+
+From version `3.58` of the SAP Cloud SDK, the OpenAPI generator has been enhanced to fail if the input specification contains keywords `anyOf` or `oneOf` in certain positions.
+(More details on these positions below)
```
Comment (artemkovalyov) [inline docs/java/features/rest/generate-rest-client.mdx:246]:
```suggestion
```

File: docs/java/features/rest/generate-rest-client.mdx [hunk 2]
```diff
@@ -237,3 +237,103 @@ The complete list of available parameters with their description is as follows:
...
+            - $ref: '#/components/schemas/SomeOtherSchema'
+```
+</details>
+
+To make the compilation to succeed, please remove these keywords (if they exist in the above mentioned positions) from the input specification and generate the client again.
+
+Please note that while removing these keywords from a property, you must also supply a meaningful replacement.
+Sometimes you may also choose to drop the property altogether for e.g, in the case of `additionalProperties`
+
+Refer to the [Swagger documentation](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) to understand more about the usage of these keywords and to derive their replacement.
```
Comment (artemkovalyov) [inline docs/java/features/rest/generate-rest-client.mdx:339]:
```suggestion
To make the compilation succeed, please remove these keywords from your specification and generate the client again.

Please note that while removing these keywords from a property, you must also supply a meaningful replacement.
Sometimes you may also choose to drop the property altogether for e.g, in the case of `additionalProperties`

Refer to the [Swagger documentation](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) to understand more about the usage of these keywords and to derive their replacement.
```

**Review by artemkovalyov — APPROVED** [review]
LGTM, thanks

**Pull request merged**